### PR TITLE
Don't convert ellipsis in markdown

### LINKED ISSUE #9904: Three dots in URLs are converted to HORIZONTAL ELLIPSIS unicode character
- Gitea version (or commit ref): 1.12.0+dev-135-g9bc841353 and 1.10.2
- Git version: N/A
- Operating system: GNU/Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes: https://try.gitea.io/WGH/Test/issues/1
  - [ ] No
  - [ ] Not relevant

## Description

URLs with three dots (like the compare links) are rendered with three dots replaced with HORIZONTAL ELLIPSIS Unicode characters, breaking links.

E.g. `https://try.gitea.io/WGH/Test/compare/master...7bc715de69acad8917132a5095d1deac46582d02`  becomes broken https://try.gitea.io/WGH/Test/compare/master%E2%80%A67bc715de69acad8917132a5095d1deac46582d02

## Workaround

Use explicit Markdown links:
```markdown
[https://try.gitea.io/WGH/Test/compare/master...7bc715de69acad8917132a5095d1deac46582d02](https://try.gitea.io/WGH/Test/compare/master...7bc715de69acad8917132a5095d1deac46582d02)
```

### PR DESCRIPTION
**Pull request by jolheiser**
Fixes #9904

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

File: modules/markup/markdown/markdown.go [hunk 1]
```diff
@@ -50,6 +50,7 @@ func RenderRaw(body []byte, urlPrefix string, wikiMarkdown bool) []byte {
 					extension.WithTypographicSubstitutions(extension.TypographicSubstitutions{
 						extension.EnDash: nil,
```
Comment (zeripath) [inline modules/markup/markdown/markdown.go]:
```suggestion
						extension.EnDash:   nil,
```

File: modules/markup/markdown/markdown.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ func RenderRaw(body []byte, urlPrefix string, wikiMarkdown bool) []byte {
...
 						extension.EmDash: nil,
```
Comment (zeripath) [inline modules/markup/markdown/markdown.go]:
```suggestion
						extension.EmDash:   nil,
```

**Comment by WGH-** [comment]
Is it the proper fix, though? I mean the bug is not in the ellipsis convertion per se, but rather that ellipsises are converted inside autolinked URLs as well.

Maybe that's the only thing we can do for the time being, but I think this warrants the upstream bug as well.

**Comment by zeripath** [comment]
It's not upstream's fault - it's ours. Really we should run this filter after we've detected all the urls - but we do that outside of the markdown renderer.

What we need to do is do the link detection from within markdown and only run our link detection on html files.

**Comment by sapk** [comment]
@jolheiser Please send backport.

**Pull request merged**